modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageExpiredRule.java

```diff
@@ -39,9 +39,12 @@ public boolean evaluate(DeliverableAndesMetadata message, ProtocolType protocolT
         //Check if destination entry has expired. Any expired message will not be delivered
         if (message.isExpired()) {
             log.warn("Message is expired. Routing Message to DLC : id= " + messageID);
+            //add the expired messages to a list for a batch delete
+            MessageExpirationWorker.expiredMessageSet.add(message);
```

Comment:
Have a method to add messages inside the worker itself. It's not good to dispose the set.